src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalUsingVarRewriter.cs

```diff
@@ -0,0 +1,113 @@
+        {
+            var localUsingVarRewriter = new LocalUsingVarRewriter();
+            return localUsingVarRewriter.Visit(statement);
+        }
+
+        public override BoundNode VisitBlock(BoundBlock node)
+        {
+            ImmutableArray<BoundStatement> statements = this.VisitList(node.Statements);
+            for (int i = 0; i < statements.Length; i++)
+            {
+                if (statements[i] is BoundLocalDeclaration localDeclaration)
+                {
+                    if (localDeclaration.LocalSymbol.IsUsing)
+                    {
+                        return LowerBoundLocalDeclarationUsingVar(node, statements);
+                    }
+                }
+                
+            }
+            return node;
+        }
+
+        private BoundBlock LowerBoundLocalDeclarationUsingVar(BoundBlock node, ImmutableArray<BoundStatement> statements)
+        {
+            int itemCount = statements.Length;
+            int firstUsingIndex = 0;
+
+            ArrayBuilder<BoundStatement> reversedStatements = ArrayBuilder<BoundStatement>.GetInstance(itemCount);
+            reversedStatements.AddRange(statements, itemCount);
+            reversedStatements.ReverseContents();
+            
+            ArrayBuilder<BoundUsingStatement> reversedUsingStatements = ArrayBuilder<BoundUsingStatement>.GetInstance();
+
+            for (int i = 0; i < itemCount; i++)
+            {
+                if (reversedStatements[i] is BoundLocalDeclaration localDeclaration && !(reversedStatements[i] is BoundReturnStatement) && SwitchBinder.ContainsUsingVariable(reversedStatements[i]))
+                {
+                    var followingStatements = GetFollowingStatements(statements, itemCount - 1 - i);
+                    firstUsingIndex = i;
+
+                    // Append inner using variable to the following statements of the previous one
+                    if (reversedUsingStatements.Count != 0)
+                    {
+                        followingStatements.Add(reversedUsingStatements[reversedUsingStatements.Count - 1]);
+                    }
+
+                    BoundBlock innerBlock = new BoundBlock(
+                            syntax: localDeclaration.Syntax,
+                            locals: ImmutableArray<LocalSymbol>.Empty,
+                            statements: followingStatements.AsImmutable()
+                            );
+
+                    BoundUsingStatement boundUsing = new BoundUsingStatement(
+                        syntax: localDeclaration.Syntax,
+                        locals: ImmutableArray<LocalSymbol>.Empty,
+                        declarationsOpt: new BoundMultipleLocalDeclarations(
+                            localDeclaration.Syntax,
+                            ImmutableArray.Create(localDeclaration)),
+                        expressionOpt: null,
+                        iDisposableConversion: Conversion.Identity,
+                        disposeMethodOpt: null,
```

Comment:
Here's the problem with the pattern using. We probably need to add something to BoundLocalDeclarations to store some of this info for later lowering, and when we're binding local declarations we'll need to check if the local is a `using` local and call into a helper to either produce an error or find the right Dispose method to call. I would suggest we could extend BoundLocalDeclarations with a BoundUsingDeclarations class.